mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt

```diff
@@ -60,56 +57,39 @@ class StoreDocumentTournament(
       }
     }
 
-  override suspend fun start(): Boolean =
-      runCatching {
-        store.transaction {
-          val tournamentDocumentReference = store.collection(Collection).document(reference.uid)
-          val current =
-              get<TournamentDocument>(tournamentDocumentReference) ?: return@transaction false
-          val adminId = current.adminId ?: return@transaction false
-          val stage = current.stage
-          val maxPlayers = current.maxPlayers ?: return@transaction false
-          val poolSize = current.poolSize ?: return@transaction false
-          val players = current.playerIds ?: return@transaction false
-          val bestOf = current.bestOf ?: return@transaction false
-
-          // Are we the tournament admin ?
-          if (adminId != user.uid) return@transaction false
-          if (stage != null) return@transaction false
-
-          val chosenPlayers = players.shuffled().take(maxPlayers)
-          val pools =
-              chosenPlayers.chunked(poolSize).map { ids ->
-                ids.map { uid ->
-                  uid to (get<ProfileDocument>(store.collection("users").document(uid))?.name ?: "")
-                }
-              }
-          val minOpponentsPerPool = pools.lastOrNull()?.size ?: 0
-
-          // TODO : Handle zero pool size.
-
-          set(tournamentDocumentReference, current.copy(stage = StagePools))
-
-          pools
-              .asSequence()
-              .mapIndexed { index, participants ->
-                PoolDocument(
-                    name = tournamentDetailsPoolName(index + 1),
-                    tournamentId = reference.uid,
-                    minOpponentsForAnyPool = minOpponentsPerPool,
-                    remainingBestOfCount = bestOf,
-                    tournamentBestOf = bestOf,
-                    tournamentAdminId = adminId,
-                    playerIds = participants.map { it.first },
-                    playerNames = participants.map { it.second },
-                )
-              }
-              .forEach { set(store.collection(PoolDocument.Collection).document(), it) }
-
-          true
+  override suspend fun start() {
+    runCatching {
+      store.transaction {
+        val tournamentDocumentReference = store.collection(Collection).document(reference.uid)
+        val current = get<TournamentDocument>(tournamentDocumentReference) ?: return@transaction
+        val adminId = current.adminId ?: return@transaction
+        val maxPlayers = current.maxPlayers ?: return@transaction
+        val poolSize = current.poolSize ?: return@transaction
+        val players = current.playerIds ?: return@transaction
+        val bestOf = current.bestOf ?: return@transaction
+        val stage = current.stage
+
+        // Are we the tournament admin ?
+        if (adminId != user.uid) return@transaction
+        if (stage != null) return@transaction
```

Comment:
Can you replace this by `val stage = current.stage ?: return@transaction` at line 70? Or they haven't the same meaning ?

Reply:
No, we actually need the opposite of what the elvis notation (`?:`) offers, we need `stage` to be `null`.